tests/test_backends.py

```diff
@@ -49,33 +73,71 @@ def test_get_backend_auto_detection_success_gguf(
     m_is_model_gguf.assert_called_once_with(tmp_gguf)
 
 
-# this test fails because the model_path is a directory but the platform is not linux
-@patch("sys.platform", "darwin")
-def test_get_backend_auto_detection_failure_vllm_dir_but_not_linux(
-    tmp_path: pathlib.Path,
+# this tests both cases where a valid and invalid safetensors model directory is supplied
+@pytest.mark.parametrize(
+    "safetensors_dir,expected",
+    [
+        (
+            "valid_safetensors_model_dir",
+            True,
+        ),
+        (
+            "invalid_safetensors_model_dir",
+            False,
+        ),
+    ],
+)
+def test_is_model_safetensors_valid(
```

Comment:
It looks like the tests are not using the `tmp_path` fixture from pytest. Please re-add that and look at other tests for reference. Also, if you do that you don't need to call `shutil.rmtree()` anymore. Thanks!

Reply:
fixed, thanks!